packages/mobile/src/components/core/TextButton.tsx

```diff
@@ -34,14 +38,42 @@ const useStyles = makeStyles(({ typography, palette }, { variant }) => {
 type TextButtonProps = TouchableOpacityProps &
   ButtonProps & {
     variant: 'primary' | 'secondary'
+    icon?: ComponentType<SvgProps>
+    iconPosition?: 'left' | 'right'
   }
 
 export const TextButton = (props: TextButtonProps) => {
-  const { title, variant, ...other } = props
+  const {
+    title,
+    variant,
+    icon: Icon,
+    iconPosition,
+    style,
+    disabled,
+    ...other
+  } = props
   const styles = useStyles({ variant })
+
+  const icon = Icon ? (
+    <Icon
+      height={18}
+      width={18}
+      fill={styles.icon.fill}
+      style={iconPosition === 'left' ? styles.iconLeft : styles.iconRight}
+    />
+  ) : null
+
   return (
-    <TouchableOpacity {...other}>
-      <Text style={styles.text}>{title}</Text>
+    <TouchableOpacity
```

Comment:
Almost seems like TextButton should just be a variant of our core Button

Reply:
yeah it definitely agreed, prop talk with julian about it when design system becomes more of a thing